test/fork/ConstantProductHelper.t.sol

```diff
@@ -0,0 +1,188 @@
+        uint256 ammUsdcInitialBalance = USDC.balanceOf(address(wethUsdcAmm));
+
+        // We send funds to the settlement contract to be able to settle the AMM
+        // rebalancing order without counterpart.
+        uint256 wethBuffer = 1_000_000 ether;
+        vm.deal(address(settlement), wethBuffer);
+        // Wrap ETH
+        vm.prank(address(settlement));
+        (bool result,) = address(WETH).call{value: wethBuffer}(hex"");
+        require(result, "ETH wrapping failed");
+        assertGe(WETH.balanceOf(address(settlement)), wethBuffer);
+
+        // Verify that the tokens are the expected ones.
+        IERC20[] memory tokens = addressVecToIerc20Vec(helper.tokens(wethUsdcAmm));
+        uint256 usdcIndex = 0;
+        uint256 wethIndex = 1;
+        assertEq(tokens.length, 2);
+        assertEq(address(tokens[usdcIndex]), address(USDC));
+        assertEq(address(tokens[wethIndex]), address(WETH));
+
+        // Prepare the price vector used in the execution of the settlement in
+        // CoW Protocol. We skew the price by ~5% towards a cheaper WETH, so
+        // that the AMM wants to buy WETH.
+        uint256[] memory prices = new uint256[](2);
+        // Note: oracle price are expressed in the same format as prices in
+        // a call to `settle`, where the  price vector is expressed so that
+        // if the first token is DAI and the second WETH then a price of 3000
+        // DAI per WETH means a price vector of [1, 3000] (if the decimals are
+        // different, as in WETH/USDC, then the atom amount is what counts).
+        prices[usdcIndex] = WETH.balanceOf(wethUsdcAmm);
+        prices[wethIndex] = USDC.balanceOf(wethUsdcAmm) * 95 / 100;
+
+        // Prepare the vector storing the one trade to be used in the
+        // settlement.
+        GPv2Trade.Data[] memory trades = new GPv2Trade.Data[](1);
+
+        // The helper generates the AMM order
+        GPv2Order.Data memory ammOrder;
+        GPv2Interaction.Data[] memory preInteractions;
+        GPv2Interaction.Data[] memory postInteractions;
+        bytes memory sig;
+        (ammOrder, preInteractions, postInteractions, sig) = helper.order(wethUsdcAmm, prices);
+        // The signature is valid for the contract but not for the
+        // settlement contract. We need to prepend the verifying contract
+        // address.
+        // TODO: update the helper so that it already includes the address in
+        // the signature.
+        sig = abi.encodePacked(wethUsdcAmm, sig);
+        trades[0] = orderToFullTrade(ammOrder, tokens, GPv2Signing.Scheme.Eip1271, sig);
+
+        // We expect a commit interaction in both pre and post interactions
+        assertEq(preInteractions.length, 1);
+        assertEq(postInteractions.length, 1);
+
+        // Because of how we changed the price, we expect to buy USDC
+        assertEq(address(ammOrder.sellToken), address(USDC));
+        assertEq(address(ammOrder.buyToken), address(WETH));
```

Comment:
this comment is misleading, if the pool is expected to **BUY** USDC, shouldn't `buyToken` be USDC?

Reply:
You're right, the comment here is incorrect. I changed it so that now it should be clearer why one is the sell token and why the other is the buy token.